packages/helper-insert-link/index.js

```diff
@@ -43,6 +43,21 @@ function injectUrl(node, value, startOffset, endOffset) {
     console.error(error);
   }
 
+  if (!el && node.textContent.includes(textMatch)) {
+    el = createLinkElement();
+
+    [...node.childNodes].forEach((child) => {
+      if (textMatch.includes(child.textContent)) {
+        if (!el.childElementCount) {
+          node.appendChild(el);
+        }
+        el.appendChild(child);
+      } else {
+        node.appendChild(child);
+      }
+    });
+  }
```

Comment:
This might be all we need to get rid of `findAndReplaceDOMText`, let's see ....

Reply:
Just tested this on #1035 and I'm able to turn the whole selection into a link now 🎉

![image](https://user-images.githubusercontent.com/831974/95642450-98beec80-0a76-11eb-90e6-b8f491de3f69.png)

Reply:
Awesome! 

In the current version of the `insert-link` package we strip out the quotation marks, to link the actual text without the surrounding quotation marks. This is mainly driven from my own desire to get a nice looking link. This seems to be the root cause for the issue you were facing. However, this change needs further testing and therefore it might be best to split this out into a separate PR